lesson2/hometask.rb

```diff
@@ -0,0 +1,26 @@
+module M
+  module SingletonMethods
+    def singleton_method
+      p Foo.singleton_methods
+    end
+  end
+  module InstanceMethods
+    def instance_method
+      p Foo.instance_methods - Object.instance_methods
+    end
+  end
+
+
+  def self.included(base)
+    base.extend SingletonMethods
+    base.include InstanceMethods
+    Foo.singleton_method
```

Comment:
Yep, it's better, but I don't like that approach.
Firstly, in the body of `M` module you directly ask about `Foo` receiver. We shouldn't do this, because `M` should know nothing about `Foo` or any other class. It's called `encapsulation`.
Secondly, you shouldn't call any method which you've defined. The `base` in `#included` should be aware of them immediately after adding their definitions. Pay attention to these two methods: `#instance_methods` and `#singleton_methods`. They could help you.

The point to self-check: your script should work the same if you add to `SingletonMethods` and `InstanceMethods` modules more than one method.